processor/src/range/mod.rs

```diff
@@ -131,6 +126,16 @@ impl RangeChecker {
     // EXECUTION TRACE GENERATION
     // --------------------------------------------------------------------------------------------
 
+    /// Converts this [RangeChecker] into an execution trace with 4 columns and the number of rows
+    /// specified by the `target_len` parameter.
+    ///
+    /// Wrapper for [`RangeChecker::into_trace_with_table`].
+    pub fn into_trace(self, target_len: usize, num_rand_rows: usize) -> RangeCheckTrace {
+        let table = self.build_8bit_lookup();
+
+        self.into_trace_with_table(table, target_len, num_rand_rows)
+    }
+
```

Comment:
similar nit: this function is only used for tests, so maybe we should mark it `#[cfg(test)]` and group it with the other one at the bottom? (I think it would just give a bit more clarity on when the 8bit lookup table is actually being built when the VM is executed)

Reply:
This function could instead be removed. I was a bit concerned to cause a breaking change so that's why I opted to keep it. But if breaking the API wouldn't be a problem here, its better we just centralize the usage into a single `into_trace` that will take the tables as arguments. wdyt?

Reply:
Ah, great point about the API. Actually I think all of these `into_trace` methods could probably be marked `pub(super)` or `pub(crate)`. They're not called outside the `processor`, and I don't see a reason why they should be. Am I missing something @bobbinth?

1. Yes, I think we could remove this wrapper and update the tests to use the method that takes the table.
2. I actually do like the current naming, since all of the other `into_trace` methods have the same signature (which doesn't take a table), but this is just a very small preference, so I think whatever you want to do with the naming is fine.

Reply:
To add to the context before @bobbinth replies; I agree with `2`. Having a different name will differentiate from the other variants that won't take a table and this will create an unmistakable domain separation (just repeating what you said haha)

Reply:
Sorry for the delay! I agree with what you guys have discussed:

1. Let's keep the name `into_trace_with_table`.
2. Let's remove `into_trace` wrapper and update the tests accordingly (if needed, we could add a helper function there).
3. I'm not sure about setting visibility to `pub(crate)` or `pub(super)`. If we wanted to keep things consistent, we'd have to apply the same to many other methods as well. Also, if I'm not mistaken, none of these components are actually exported from the crate - so, they are not a part of the crate's public interface anyway.

Reply:
Confirmed, these are not exported. I kept them as public, so we do a more general refactor in the future if we want to introduce `pub(crate)`